src/test/java/io/cdap/plugin/servicenow/source/ServiceNowSourceConfigTest.java

```diff
@@ -432,40 +445,150 @@ public void testEndDateLessThanStartDate() {
     Assert.assertEquals(2, collector.getValidationFailures().size());
   }
 
-  private ServiceNowSourceConfig withServiceNowValidationMock(ServiceNowSourceConfig config,
-                                                              FailureCollector collector) {
-    ServiceNowSourceConfig spy = Mockito.spy(config);
-    Mockito.doNothing().when(spy).validateServiceNowConnection(collector);
-    return spy;
-  }
-
   @Test
-  public void testValidateTableNames() {
+  public void testStartDateInvalidEndDateInvalid() {
     MockFailureCollector collector = new MockFailureCollector();
-    String tableNames = "tablenames";
-    ServiceNowMultiSourceConfig config = ServiceNowSourceConfigHelper.newConfigBuilder()
-      .setTableNames(tableNames)
-      .buildMultiSource();
-    config.validateTableNames(collector);
+    ServiceNowSourceConfig config = withServiceNowValidationMock(ServiceNowSourceConfigHelper.newConfigBuilder()
+      .setClientId(ServiceNowSourceConfigHelper.TEST_CLIENT_ID)
+      .setClientSecret(ServiceNowSourceConfigHelper.TEST_CLIENT_SECRET)
+      .setRestApiEndpoint(ServiceNowSourceConfigHelper.TEST_API_ENDPOINT)
+      .setUser(ServiceNowSourceConfigHelper.TEST_USER)
+      .setPassword(ServiceNowSourceConfigHelper.TEST_PASSWORD)
+      .setQueryMode("Table")
+      .setTableName("ast-contract")
+      .setStartDate("2019")
+      .setEndDate("2020")
+      .build(), collector);
 
-    Assert.assertEquals(1, collector.getValidationFailures().size());
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+    } catch (ValidationException e) {
+      Assert.assertEquals(ServiceNowConstants.PROPERTY_START_DATE, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+
+    Assert.assertEquals(3, collector.getValidationFailures().size());
```

Comment:
why do we get 3 validation failures? Let's have a separate test case for each config that we are testing.

Reply:
This test case was created to cover the use case when both the start date as well as the end dates were invalid.

The individual test case for i[nvalid start date](https://github.com/data-integrations/servicenow-plugins/blob/a77cbb135841c8ea7c9fe529c0275ddcfaacf4bc/src/test/java/io/cdap/plugin/servicenow/source/ServiceNowSourceConfigTest.java#L371) and [invalid end date](https://github.com/data-integrations/servicenow-plugins/blob/a77cbb135841c8ea7c9fe529c0275ddcfaacf4bc/src/test/java/io/cdap/plugin/servicenow/source/ServiceNowSourceConfigTest.java#L397) was already present before.

Reply:
I see. Why do we have 3 validation errors if start date and end date are invalid (2 configs). Can you please add a comment in the code to clarify.

Reply:
A mock was missing for a newly added method : `ServiceNowBaseSourceConfig.validateTable`
After adding this mock in the `ServiceNowSourceConfigTest.withServiceNowValidationMock`, I've updated the assertions for the JUnits.
For this test case: `testStartDateInvalidEndDateInvalid`, I've updated the assertion from three to two (one for invalid start date and one for invalid end date) 
[2d8729d](https://github.com/data-integrations/servicenow-plugins/pull/15/commits/2d8729d9263201d6896f4af6fb44b14d4ee53be2)